App/Sagas/HandleSharePhotoRequest.ts

```diff
@@ -0,0 +1,56 @@
+import { call, put } from 'redux-saga/effects'
+import { ActionType } from 'typesafe-actions'
+import RNFS from 'react-native-fs'
+
+import { uploadFile } from './UploadFile'
+import TextileNode from '../../TextileNode'
+import { SharedImage, AddResult } from '../Models/TextileTypes'
+import TextileNodeActions from '../Redux/TextileNodeRedux'
+import ProcessingImagesActions from '../Redux/ProcessingImagesRedux'
+import UIActions from '../Redux/UIRedux'
+
+export function * handleSharePhotoRequest(action: ActionType<typeof UIActions.sharePhotoRequest>) {
+  const { image, threadId, comment } = action.payload
+  if (!image || !threadId) {
+    return
+  }
+  if (typeof image === 'string') {
+    yield call(shareWalletImage, image, threadId, comment)
+  } else {
+    yield call(addAndUploadImage, image, threadId, comment)
+  }
+}
+
+function * shareWalletImage (id: string, threadId: string, comment?: string) {
+  try {
+    // TODO: Insert some state into the processing photos redux in case this takes long or fails
+    const blockId: string = yield call(TextileNode.sharePhotoToThread, id, threadId, comment)
+    yield put(TextileNodeActions.getPhotoHashesRequest(threadId))
+  } catch (error) {
+    yield put(UIActions.imageSharingError(error))
+  }
+}
+
+function * addAndUploadImage (image: SharedImage, threadId: string, comment?: string) {
+  try {
+    yield put(ProcessingImagesActions.insertAddingImage(image, threadId, comment))
+    const path = image.uri.replace('file://', '')
```

Comment:
didn't notice, is the Android path variation being handled way back somewhere now?

Reply:
Ah not sure. What does the android `uri` look like, do you know?

Reply:
the picker payload includes a `.path` you should use instead of `uri` i think